### PR TITLE
[WFLY-12206] Dependencies versions bump

### PR DESCRIPTION
**Pull request by rsvoboda**
Dependencies versions bump - micro versions bump
https://issues.jboss.org/browse/WFLY-12206

Please make sure your PR meets the following requirements:
- [x] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue(s)
- [x] Pull Request contains description of the issue(s)
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted

### CODE REVIEW

**Comment by darranl** [comment]
TBH with this number of version bumps it is better to split them into smaller commits, with all in one commit it will make it impossible in future to git-bisect to identify the cause of an issue.

Also have these all been checked to ensure they are only bringing in bug fixes and no new features?  For each we should really see the release notes but if they have been manually checked maybe that will be sufficient.

File: pom.xml [hunk 1]
```diff
@@ -309,19 +309,19 @@
...
+        <version.org.cryptacular>1.2.3</version.org.cryptacular>
         <version.org.eclipse.jdt.core.compiler>4.6.1</version.org.eclipse.jdt.core.compiler>
         <version.org.eclipse.microprofile.config.api>1.3</version.org.eclipse.microprofile.config.api>
         <version.org.eclipse.microprofile.health.api>1.0</version.org.eclipse.microprofile.health.api>
         <version.org.eclipse.microprofile.metrics.api>1.1</version.org.eclipse.microprofile.metrics.api>
         <version.org.eclipse.microprofile.opentracing>1.1</version.org.eclipse.microprofile.opentracing>
         <version.org.eclipse.microprofile.rest.client.api>1.0.1</version.org.eclipse.microprofile.rest.client.api>
-        <version.org.eclipse.yasson>1.0.3</version.org.eclipse.yasson>
-        <version.org.eclipselink.version>2.7.3</version.org.eclipselink.version>
+        <version.org.eclipse.yasson>1.0.4</version.org.eclipse.yasson>
```
Comment (jamezp) [inline pom.xml]: We can't do this one yet.

**Review by jamezp — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -257,20 +257,20 @@
...
-        <version.io.opentracing.tracerresolver>0.1.5</version.io.opentracing.tracerresolver>
+        <version.io.opentracing.tracerresolver>0.1.8</version.io.opentracing.tracerresolver>
         <version.io.opentracing.servlet>0.1.0</version.io.opentracing.servlet>
-        <version.io.reactivex.rxjava>2.2.2</version.io.reactivex.rxjava>
+        <version.io.reactivex.rxjava>2.2.9</version.io.reactivex.rxjava>
         <version.io.prometheus.simpleclient>0.6.0</version.io.prometheus.simpleclient>
-        <version.io.smallrye.smallrye-config>1.3.4</version.io.smallrye.smallrye-config>
-        <version.io.smallrye.smallrye-health>1.0.2</version.io.smallrye.smallrye-health>
+        <version.io.smallrye.smallrye-config>1.3.7</version.io.smallrye.smallrye-config>
+        <version.io.smallrye.smallrye-health>1.0.3</version.io.smallrye.smallrye-health>
```
Comment (jamezp) [inline pom.xml]: Looks like this now requires a dependency on `org.jboss.logging`.

File: pom.xml (hunk 1 above)
Comment (asoldano) [reply pom.xml]: @jamezp Can I ask for more details? The same yasson upgrade would come with the resteasy 3.8.0.Final upgrade these days.

Comment (asoldano) [reply pom.xml]: Nevermind, just read on the mailing list...

File: pom.xml [hunk 3]
```diff
@@ -309,19 +309,19 @@
...
         <version.org.eclipse.jdt.core.compiler>4.6.1</version.org.eclipse.jdt.core.compiler>
         <version.org.eclipse.microprofile.config.api>1.3</version.org.eclipse.microprofile.config.api>
         <version.org.eclipse.microprofile.health.api>1.0</version.org.eclipse.microprofile.health.api>
         <version.org.eclipse.microprofile.metrics.api>1.1</version.org.eclipse.microprofile.metrics.api>
         <version.org.eclipse.microprofile.opentracing>1.1</version.org.eclipse.microprofile.opentracing>
         <version.org.eclipse.microprofile.rest.client.api>1.0.1</version.org.eclipse.microprofile.rest.client.api>
-        <version.org.eclipse.yasson>1.0.3</version.org.eclipse.yasson>
-        <version.org.eclipselink.version>2.7.3</version.org.eclipselink.version>
+        <version.org.eclipse.yasson>1.0.4</version.org.eclipse.yasson>
+        <version.org.eclipselink.version>2.7.4</version.org.eclipselink.version>
```
Comment (scottmarlow) [inline pom.xml]: I don't think that the EclipseLink upgrade is needed but if it happens, that is fine.

File: pom.xml (hunk 1 above)
Comment (scottmarlow) [reply pom.xml]: Note that GlassFish [5.1](https://github.com/eclipse-ee4j/glassfish/blob/5.1.0-BRANCH/appserver/pom.xml#L102)/[master](https://github.com/eclipse-ee4j/glassfish/blob/master/appserver/pom.xml#L102) branches are also still on yasson 1.0.3

File: pom.xml [hunk 4]
```diff
@@ -239,16 +239,16 @@
...
+        <version.com.google.code.gson>2.8.5</version.com.google.code.gson>
         <version.com.google.guava>25.0-jre</version.com.google.guava>
         <version.com.h2database>1.4.193</version.com.h2database>
         <version.com.microsoft.azure>6.1.0</version.com.microsoft.azure>
         <version.com.squareup.okhttp3>3.9.0</version.com.squareup.okhttp3>
         <version.com.squareup.okio>1.13.0</version.com.squareup.okio>
         <version.com.sun.faces>2.3.9.SP02</version.com.sun.faces>
-        <version.com.sun.istack>3.0.7</version.com.sun.istack>
-        <version.com.sun.xml.fastinfoset>1.2.13</version.com.sun.xml.fastinfoset>
+        <version.com.sun.istack>3.0.8</version.com.sun.istack>
```
Comment (jimma) [inline pom.xml]: CXF master is still with using istack 3.0.7.

File: pom.xml [hunk 5]
```diff
@@ -309,19 +309,19 @@
         <version.org.apache.ws.security>2.2.3</version.org.apache.ws.security>
         <version.org.apache.ws.xmlschema>2.2.4</version.org.apache.ws.xmlschema>
         <version.org.apache.xalan>2.7.1.jbossorg-4</version.org.apache.xalan>
-        <version.org.bouncycastle>1.60</version.org.bouncycastle>
+        <version.org.bouncycastle>1.62</version.org.bouncycastle>
```
Comment (jimma) [inline pom.xml]: JBossWS/CXF is still using 1.60. Let's upgrade this version in CXF first to see if all work, then we back to upgrade WFLY.

**Comment by bstansberry** [comment]
I put the 'hold' label on this because it's basically functioning as a discussion place for proposed upgrades. Any actual upgrades will be done via separate, focused PRs with associated component upgrade JIRAs.

File: pom.xml (hunk 3 above)
Comment (bstansberry) [reply pom.xml]: @scottmarlow If you want this upgrade please make it happen separately.

File: pom.xml [hunk 6]
```diff
@@ -257,20 +257,20 @@
...
         <version.io.opentracing.jaxrs2>0.1.7</version.io.opentracing.jaxrs2>
-        <version.io.opentracing.tracerresolver>0.1.5</version.io.opentracing.tracerresolver>
+        <version.io.opentracing.tracerresolver>0.1.8</version.io.opentracing.tracerresolver>
         <version.io.opentracing.servlet>0.1.0</version.io.opentracing.servlet>
-        <version.io.reactivex.rxjava>2.2.2</version.io.reactivex.rxjava>
+        <version.io.reactivex.rxjava>2.2.9</version.io.reactivex.rxjava>
         <version.io.prometheus.simpleclient>0.6.0</version.io.prometheus.simpleclient>
-        <version.io.smallrye.smallrye-config>1.3.4</version.io.smallrye.smallrye-config>
-        <version.io.smallrye.smallrye-health>1.0.2</version.io.smallrye.smallrye-health>
+        <version.io.smallrye.smallrye-config>1.3.7</version.io.smallrye.smallrye-config>
```
Comment (jmesnil) [inline pom.xml]: looks good.

**Pull request closed without merging**